WordPress/MediaEditor/MediaEditor.swift

```diff
@@ -1,103 +1,67 @@
 import UIKit
-import TOCropViewController
 
-public class MediaEditor: NSObject {
+public class MediaEditor: UINavigationController {
+    static var capabilities: [MediaEditorCapability.Type] = [MediaEditorCrop.self]
 
-    let cropViewControllerFactory: (UIImage) -> TOCropViewController
+    var hub: MediaEditorHub
 
-    private lazy var cropViewController: TOCropViewController = {
-        return cropViewControllerFactory(image)
-    }()
-    private var onFinishEditing: ((UIImage, [MediaEditorOperation]) -> ())?
-    private var onCancel: (() -> ())?
+    var image: UIImage?
 
-    public let image: UIImage
+    var onFinishEditing: ((UIImage, [MediaEditorOperation]) -> ())?
 
-    public var doneTextButton: UIButton {
-        return cropViewController.toolbar.doneTextButton
-    }
-
-    public var cancelTextButton: UIButton {
-        return cropViewController.toolbar.cancelTextButton
-    }
-
-    public var resetButton: UIButton {
-        return cropViewController.toolbar.resetButton
-    }
-
-    public var doneIconButton: UIButton {
-        return cropViewController.toolbar.doneIconButton
-    }
+    var onCancel: (() -> ())?
 
-    public var cancelIconButton: UIButton {
-        return cropViewController.toolbar.cancelIconButton
-    }
-
-    public var rotateClockwiseButton: UIButton? {
-        return cropViewController.toolbar.rotateClockwiseButton
-    }
-
-    public var rotateCounterclockwiseButton: UIButton? {
-        return cropViewController.toolbar.rotateCounterclockwiseButton
-    }
-
-    public var rotateCounterclockwiseButtonHidden: Bool {
-        get {
-            return cropViewController.toolbar.rotateCounterclockwiseButtonHidden
-        }
+    private(set) var currentCapability: MediaEditorCapability?
 
-        set {
-            cropViewController.toolbar.rotateCounterclockwiseButtonHidden = newValue
+    public var styles: MediaEditorStyles = [:] {
+        didSet {
+            currentCapability?.apply(styles: styles)
         }
     }
 
-    public var rotateClockwiseButtonHidden: Bool {
-        get {
-            return cropViewController.toolbar.rotateClockwiseButtonHidden
-        }
-
-        set {
-            cropViewController.toolbar.rotateClockwiseButtonHidden = newValue
-        }
+    init(_ image: UIImage) {
+        self.image = image
+        hub = MediaEditorHub()
+        super.init(rootViewController: hub)
+        presentIfSingleImageAndCapability()
+        modalTransitionStyle = .crossDissolve
+        modalPresentationStyle = .fullScreen
+        navigationBar.isHidden = true
     }
 
-    public init(cropViewControllerFactory: @escaping (UIImage) -> TOCropViewController = TOCropViewController.init, image: UIImage) {
-        self.cropViewControllerFactory = cropViewControllerFactory
-        self.image = image
-        super.init()
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
```

Comment:
I expected `super.init(coder: aDecoder)`

Reply:
Done in 7461d1b913edca1fd9728b30dc06dea66b1417cf